src/SqlAsyncCollector.cs

```diff
@@ -543,7 +570,12 @@ public static async Task<TableInformation> RetrieveTableInformationAsync(SqlConn
                 // Match SQL Primary Key column names to POCO field/property objects. Ensure none are missing.
                 StringComparison comparison = caseSensitive ? StringComparison.Ordinal : StringComparison.OrdinalIgnoreCase;
                 IEnumerable<MemberInfo> primaryKeyFields = typeof(T).GetMembers().Where(f => primaryKeys.Any(k => string.Equals(k.Name, f.Name, comparison)));
-                IEnumerable<string> primaryKeysFromPOCO = primaryKeyFields.Select(f => f.Name);
+                // When T is JObject, primaryKeyFields cannot be used to get the column names, so use the passed in columnNames variable
```

Comment:
You should refactor this to have the column names always be passed in - and then for POCO objects it would just be the list of properties from the object. Having to sometimes use the parameter and sometimes not is confusing and could easily result in logic mistakes.